tests/lib/rules/no-unsafe-negation.js

```diff
@@ -29,7 +48,16 @@ ruleTester.run("no-unsafe-negation", rule, {
         "a instanceof b",
         "a instanceof b === false",
         "!(a instanceof b)",
-        "(!a) instanceof b"
+        "(!a) instanceof b",
+        "if (! a < b) {}",
+        "while (! a > b) {}",
+        "foo = ! a <= b;",
+        "foo = ! a >= b;",
+        {
+            code: "foo = (!a) >= b;",
+            options: [{ enforceForOrderingRelations: true }],
+            errors: [unexpectedMoreThanOrEqualOperatorError]
```

Comment:
This is a very good test case for the built-in exception to the rule, but the `errors` property should be removed.

It would be also good to have some tests where the option is `true` and the code is valid, similar to tests for `instanceof` (e,g., without negation, or where the whole expression is negated not just the left operand)